hibernate-core/src/main/java/org/hibernate/type/descriptor/java/LocalDateJavaDescriptor.java

```diff
@@ -112,7 +114,15 @@ public LocalDate fromString(String string) {
 
 		if ( Date.class.isInstance( value ) ) {
 			if ( java.sql.Date.class.isInstance( value ) ) {
-				return ((java.sql.Date) value).toLocalDate();
+				final java.sql.Date date = (java.sql.Date) value;
+				// If the JDBC timezone is given, the date might have been adjusted accordingly if we have a different default timezone
+				if ( options.getJdbcTimeZone() != null ) {
```

Comment:
Shouldn't the `hibernate.jdbc.time_zone` already take care of this? If we provide this configuration property, then the `org.hibernate.type.descriptor.sql.TimestampTypeDescriptor` should read the Timestamp using the provided time zone.